packages/react/navigation-menu/src/NavigationMenu.tsx

```diff
@@ -354,8 +328,10 @@ interface NavigationMenuItemProps extends PrimitiveListItemProps {
 const NavigationMenuItem = React.forwardRef<NavigationMenuItemElement, NavigationMenuItemProps>(
   (props: ScopedProps<NavigationMenuItemProps>, forwardedRef) => {
     const { __scopeNavigationMenu, value: valueProp, ...itemProps } = props;
-    const autoValue = React.useMemo(() => String(Math.random()), []);
-    const value = valueProp || autoValue;
+    const autoValue = useId();
+    // We need to provide an initial deterministic value as `useId` will return
+    // empty string on the first render and we don't want to match our internal "closed" value.
+    const value = valueProp || autoValue || 'LEGACY_REACT_AUTO_VALUE';
```

Comment:
Couldn't pass to `useId` here as that would always use the ID in place of the generated version (so all items would be the same). Instead we want to use the deterministic fallback for the first render only.

Reply:
Ah yeah good point, otherwise it always uses the one you provide as an argument.